lib/dotcom/system_status/subway_cache.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Dotcom.SystemStatus.SubwayCache do
+  @moduledoc """
+  A GenServer that periodically computes subway status and broadcasts changes when
+  they happen
+  """
+
+  use GenServer
+
+  alias Dotcom.SystemStatus
+  alias Dotcom.SystemStatus.SubwayCache.Behaviour
+
+  @behaviour Behaviour
+
+  # Client
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  @impl Behaviour
+  def subway_status() do
+    GenServer.call(__MODULE__, :subway_status)
+  end
+
+  # Server
+
+  @impl true
+  def init(_opts) do
+    Phoenix.PubSub.subscribe(Dotcom.PubSub, "alerts")
```

Comment:
nit: I think you can use `DotcomWeb.Endpoint.subscribe("alerts")` here, as I believe the endpoint has all these (subscribe/broadcast/etc) functions available